r/configure.win

```diff
@@ -39,8 +39,9 @@ if [ -d "windows/arrow-$VERSION" ]; then
 else
   # It's possible that the version of the libarrow binary is not identical to the
   # R version, e.g. if the R build is a patch release, so find what the dir is
-  # actually called:
-  RWINLIB="../windows/$(ls windows/ | grep ^arrow-)"
+  # actually called. If there is more than one version present, use the one
+  # with the highest version:
+  RWINLIB="../windows/$(ls windows/ | grep ^arrow- | tail -n 1)"
```

Comment:
`ls` sorts in ascending order by name, so `tail -n 1` gives the highest version string